lib/ansible/runner.rb

```diff
@@ -0,0 +1,29 @@
+module Ansible
+  class Runner
+    class << self
+      def run(env_vars, extra_vars, playbook_path)
+        run_via_cli(env_vars, extra_vars, playbook_path)
+      end
+
+      private
+
+      def run_via_cli(env_vars, extra_vars, playbook_path)
+        result = %x(#{format_env_vars(env_vars)} #{ansible_command.shellescape} #{format_extra_vars(extra_vars)} #{playbook_path.to_s.shellescape})
```

Comment:
I think it would be better to use AwesomeSpawn here for building the CLI and executing.  This way, you don't have to do any of the shell escaping or env handling yourself as AwesomeSpawn already handles it, and this can focus on just using normal hashes.  AwesomeSpawn can also handle non-zero exit codes.

```ruby
def run_via_cli(env_vars, extra_vars, playbook_path)
  result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
  JSON.parse(result.output)
end

# no need for all of the format methods here
```

Reply:
the AwesomeSpawn is awesome :-)

Reply:
:laughing: